Utilities/UploadDataToDataBase.ps1

```diff
@@ -26,55 +26,46 @@ Param
-
-try
-{
-	$ExitCode = 1
-	$XmlSecrets = ""
-	if (![String]::IsNullOrEmpty($AzureSecretsFile) -and (Test-Path -Path $AzureSecretsFile) -eq $true) {
-		$XmlSecrets = ([xml](Get-Content $AzureSecretsFile))
-	} else {
-		Write-Host "Error: Please provide value for -AzureSecretsFile"
-	}
-	if (![String]::IsNullOrEmpty($XmlSecrets)) {
-		$dataSource = $XmlSecrets.secrets.DatabaseServer
-		$dbuser = $XmlSecrets.secrets.DatabaseUser
-		$dbpassword = $XmlSecrets.secrets.DatabasePassword
-		$database = $XmlSecrets.secrets.DatabaseName
+$ExitCode = 1
+$XmlSecrets = ""
+if (![String]::IsNullOrEmpty($AzureSecretsFile) -and (Test-Path -Path $AzureSecretsFile)) {
+	$XmlSecrets = ([xml](Get-Content $AzureSecretsFile))
+} else {
+	Write-Host "Error: Please provide value for -AzureSecretsFile"
+}
+if ((![String]::IsNullOrEmpty($XmlSecrets)) -and (![String]::IsNullOrEmpty($XmlSecrets.secrets))) {
+	$dataSource = $XmlSecrets.secrets.DatabaseServer
+	$dbuser = $XmlSecrets.secrets.DatabaseUser
+	$dbpassword = $XmlSecrets.secrets.DatabasePassword
+	$database = $XmlSecrets.secrets.DatabaseName
 
-		if ($dataSource -and $dbuser -and $dbpassword -and $database) {
-			try {
-				Write-Host "Info: SQLQuery:  $SQLQuery"
-				$connectionString = "Server=$dataSource;uid=$dbuser; pwd=$dbpassword;Database=$database;Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;"
-				$connection = New-Object System.Data.SqlClient.SqlConnection
-				$connection.ConnectionString = $connectionString
-				$connection.Open()
-				$command = $connection.CreateCommand()
-				$command.CommandText = $SQLQuery
-				$null = $command.executenonquery()
-				$connection.Close()
-				$ExitCode = 0
-				Write-Host "Info: Uploading data to database :  done!!!"
-			} catch {
-				Write-Host "Error: Uploading data to database :  ERROR"
-				$line = $_.InvocationInfo.ScriptLineNumber
-				$script_name = ($_.InvocationInfo.ScriptName).Replace($PWD,".")
-				$ErrorMessage =  $_.Exception.Message
-				Write-Host "Error: EXCEPTION : $ErrorMessage"
-				Write-Host "Error: Source : Line $line in script $script_name."
-			}
-		} else {
-			Write-Host "Error: Database details are not provided. Data will not be uploaded to database!!!"
+	if ($dataSource -and $dbuser -and $dbpassword -and $database) {
+		try {
+			Write-Host "Info: SQLQuery:  $SQLQuery"
+			$connectionString = "Server=$dataSource;uid=$dbuser; pwd=$dbpassword;Database=$database;Encrypt=yes;TrustServerCertificate=no;Connection Timeout=30;"
+			$connection = New-Object System.Data.SqlClient.SqlConnection
+			$connection.ConnectionString = $connectionString
+			$connection.Open()
+			$command = $connection.CreateCommand()
+			$command.CommandText = $SQLQuery
+			$null = $command.executenonquery()
+			$connection.Close()
+			$ExitCode = 0
+			Write-Host "Info: Uploading data to database :  done!!!"
+		} catch {
+			Write-Host "Error: Uploading data to database :  ERROR"
```

Comment:
If you want to improve the code, then you can keep the original try-catch block and call here throw statement. The outer catch can handle the exception.

Reply:
the original outer try-catch is useless, as I explained in the last comment. we just improve the inner try-catch logic to handle and release necessary resources.